extensions/logging/README.md

```diff
@@ -0,0 +1,190 @@
+# @loopback/extension-logging
+
+This module contains a component provides logging facilities based on
+[Winston](https://github.com/winstonjs/winston) and
+[Fluentd](https://github.com/fluent/fluent-logger-node).
+
+## Stability: :warning:Experimental:warning:
+
+> Experimental packages provide early access to advanced or experimental
+> functionality to get community feedback. Such modules are published to npm
+> using `0.x.y` versions. Their APIs and functionality may be subject to
+> breaking changes in future releases.
+
+## Architecture overview
+
+![logging-component](logging-component.png)
+
+## Installation
+
+```sh
+npm install --save @loopback/extension-logging
+```
+
+## Basic use
+
+The component should be loaded in the constructor of your custom Application
+class.
+
+Start by importing the component class:
+
+```ts
+import {LoggingComponent} from '@loopback/extension-logging';
+```
+
+In the constructor, add the component to your application:
+
+```ts
+this.component(LoggingComponent);
+```
+
+Now your application can add a controller as follows to leverage the logging
+facilities:
+
+```ts
+import {inject} from '@loopback/context';
+import {Logger, log} from '@loopback/extension-logging';
+import {get, param} from '@loopback/rest';
+
+class MyController {
+  // Inject a winston logger
+  @inject(LoggingBindings.WINSTON_LOGGER)
+  private logger: Logger;
+
+  // http access is logged by a global interceptor
+  @get('/greet/{name}')
+  // log the `greet` method invocations
+  @log()
+  greet(@param.path.string('name') name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @get('/hello/{name}')
+  hello(@param.path.string('name') name: string) {
+    // Use the winston logger explicitly
+    this.logger.log('info', `greeting ${name}`);
+    return `Hello, ${name}`;
+  }
+}
+```
+
+## Configure the logging component
+
+The logging component can be configured as follows:
+
+```ts
+app.configure(LoggingBindings.COMPONENT).to({
+  enableFluent: false, // default to true
+  enableHttpAccessLog: true, // default to true
+});
+```
```

Comment:
I have mixed feelings about the design where we call `this.component()` in one place and then `app.configure` in another. Personally, I would expect to find the configuration in the same place where the component is mounted, i.e.

```ts
this.component(LoggingComponent, {
   enableFluent: false, // default to true
   enableHttpAccessLog: true, // default to true
});
```

Implementation wise, I can imagine `this.component` calling `app.configure` under the hood.

The biggest issue I see is how to add type safety, to let the compiler understand what configuration options the component allows. IIUC, this problem is not solved for `app.configure` either, therefore it's out of scope of this pull request. I just want to point it out.

@strongloop/loopback-next thoughts?

On the second thought, I think there is even more important aspect we should consider: which part of the configuration are specific to deployment environment (dev/staging/production) and which are wiring different code bits together (thing of DI)? The former is typically provided from outside of the application, for example consider the port where the REST server will listen on.

In this case, I think both `enableFluent` and `enableHttpAccessLog` option are affecting which code bits are mixed into the application, therefore they should be hard-coded by the app, similarly to how we are hard-coding dependency wirings.

On the other hand, I feel the configuration of the transport and possibly logger is something to configure from outside of the app, depending on the deployment environment.

Anyhow, I guess this can be left out of scope of this initial pull request, because the logging extension is clearly labelled as experimental.